### DISCUSSION TITLE
npm run docker:run Error: listen EADDRINUSE: address already in use 127.0.0.1:3000

### DISCUSSION THREAD

**Discussion by bostjanpisler** [loopbackio/loopback-next#8416] [LB3 Migration]
When trying to run tests of lb3 app mounted inside lb4 using provided Dockerfile I get this error:

```bash
Error: listen EADDRINUSE: address already in use 127.0.0.1:3000
      at Server.setupListenHandle [as _listen2] (node:net:1330:16)
      at listenInCluster (node:net:1378:12)
      at doListen (node:net:1516:7)
      at processTicksAndRejections (node:internal/process/task_queues:84:21)
}
```

Locally tests run normally.

How to reproduce:
`lb4 example lb3-application`

Add Dockerfile
```
# Check out https://hub.docker.com/_/node to select a new base image
FROM node:16-slim

# Set to a non-root built-in user `node`
USER node

# Create app directory (with user `node`)
RUN mkdir -p /home/node/app

WORKDIR /home/node/app

# Install app dependencies
# A wildcard is used to ensure both package.json AND package-lock.json are copied
# where available (npm@5+)
COPY --chown=node package*.json ./

RUN npm install

# Bundle app source code
COPY --chown=node . .

RUN npm run build

# Bind to all network interfaces so that it can be mapped to the host OS
ENV HOST=0.0.0.0 PORT=3000

EXPOSE ${PORT}
CMD [ "node", "." ]
```

Run tests using docker
`docker build -f Dockerfile -t lb3-application:test . && docker run -p 3000:3000 --rm -it lb3-application:test npm run test`

**Comment by bostjanpisler** [comment]
@frbuceta would you maybe know how to solve this?

**Reply by frbuceta** [reply]
It fails me locally too. I am going to check what is happening and why it does not fail in the monorepo. I keep you informed

**Reply by frbuceta** [reply]
@BostjanPisler You can change the port argument to 0 and the tests will stop fine. Can you check it?

https://github.com/loopbackio/loopback-next/blob/3d6dd2ef3ae55a65f907ee2da99ca57d285fd061/examples/lb3-application/src/server.ts#L58
